File: CppFramework/ConsoleGameLib/src/GameConsole.cpp
#include <ConsoleGameLib/GameConsole.h>
#include <Implementation/GConsole.h>

void CGL::Console::clearBuffer() {
	// Get and check console pointer
	auto consolePointer = CGL_internal::GConsole::get();
	if (consolePointer) {
		// Redirect call
		consolePointer->clearBuffer();
	}
}

void CGL::Console::present() {
	// Get and check console pointer
	auto consolePointer = CGL_internal::GConsole::get();
	if (consolePointer) {
		// Redirect call
		consolePointer->present();
	}
}

void CGL::Console::putChar(unsigned int x, unsigned int y, char c) {
	// Get and check console pointer
	auto consolePointer = CGL_internal::GConsole::get();
	if (consolePointer) {
		// Redirect call
		consolePointer->putChar(x, y, c);
	}
}

void CGL::Console::putString(unsigned int x, unsigned int y, const char* string, bool wrap) {
	// Get and check console pointer
	auto consolePointer = CGL_internal::GConsole::get();
	if (consolePointer) {
		// Redirect call
		consolePointer->putString(x, y, string, wrap);
	}
}

File: CppFramework/HangmanScript/CGL/GameConsole.cs
using System;
using System.Runtime.CompilerServices;

namespace CGL {
    public class GameConsole {
        [MethodImplAttribute(MethodImplOptions.InternalCall)]
        public extern static void clearBuffer();

        [MethodImplAttribute(MethodImplOptions.InternalCall)]
        public extern static void present();

        [MethodImplAttribute(MethodImplOptions.InternalCall)]
        public extern static void putChar(UInt16 x, UInt16 y, char c);

        [MethodImplAttribute(MethodImplOptions.InternalCall)]
        public extern static void putString(UInt16 x, UInt16 y, string str, bool wrap = true);
    }
}
